File: src/main/java/com/nerdearla/workshop/model/PaymentOperation.java
package com.nerdearla.workshop.model;

import com.nerdearla.workshop.dto.payment.PaymentRequest;

public class PaymentOperation {

    private final PaymentRequest paymentRequest;
    private String paymentId;
    private QR qr;
    private Buyer buyer;
    private Seller seller;
    private BuyerPaymentMethod buyerPaymentMethod;

    public PaymentOperation(PaymentRequest paymentRequest) {
        this.paymentRequest = paymentRequest;
    }

    public PaymentRequest getPaymentRequest() {
        return paymentRequest;
    }

    public String getPaymentId() {
        return paymentId;
    }

    public void setPaymentId(String paymentId) {
        this.paymentId = paymentId;
    }

    public Buyer getBuyer() {
        return buyer;
    }

    public void setBuyer(Buyer buyer) {
        this.buyer = buyer;
    }

    public Seller getSeller() {
        return seller;
    }

    public void setSeller(Seller seller) {
        this.seller = seller;
    }

    public BuyerPaymentMethod getPaymentMethod() {
        return buyerPaymentMethod;
    }

    public void setPaymentMethod(BuyerPaymentMethod buyerPaymentMethod) {
        this.buyerPaymentMethod = buyerPaymentMethod;
    }

    public QR getQr() {
        return qr;
    }

    public void setQr(QR qr) {
        this.qr = qr;
    }
}

File: src/main/java/com/nerdearla/workshop/model/PaymentOperation.kt
package com.nerdearla.workshop.model

import com.nerdearla.workshop.dto.buyer.Buyer
import com.nerdearla.workshop.dto.buyer.paymentMethod.BuyerPaymentMethod
import com.nerdearla.workshop.dto.fraud.FraudValidationResponse
import com.nerdearla.workshop.dto.payment.PaymentMethodData
import com.nerdearla.workshop.dto.payment.PaymentRequest
import com.nerdearla.workshop.dto.payment.TerminalData
import com.nerdearla.workshop.dto.qr.QR
import com.nerdearla.workshop.dto.seller.Seller

data class InitialOperation(
    val qrId: String,
    val buyerId: String,
    val identification: String,
    val gender: String,
    val sellerId: String,
    val amount: Double,
    val installments: Int = 1,
    val terminalData: TerminalData,
    val paymentMethodData: PaymentMethodData
)

data class FullOperation(
    val paymentId: String,
    val qr: QR,
    val buyerPaymentMethod: BuyerPaymentMethod,
    val amount: Double,
    val installments: Int = 1,
    val seller: Seller,
    val buyer: Buyer,
    val terminalData: TerminalData
)

data class NotFraudulentOperation(
    val paymentId: String,
    val qr: QR,
    val buyerPaymentMethod: BuyerPaymentMethod,
    val amount: Double,
    val installments: Int = 1,
    val seller: Seller,
    val buyer: Buyer,
    val terminalData: TerminalData,
    val fraudResponse: FraudValidationResponse
)

data class AuthorizedOperation(
    val paymentId: String,
    val qr: QR,
    val buyerPaymentMethod: BuyerPaymentMethod,
    val amount: Double,
    val installments: Int = 1,
    val seller: Seller,
    val buyer: Buyer,
    val terminalData: TerminalData,
    val fraudResponse: FraudValidationResponse,
    val authorization: PaymentAuthorization
)
